haystack/nodes/llm/prompt_node.py

```diff
@@ -0,0 +1,643 @@
+import json
+import logging
+from abc import ABC, abstractmethod
+from collections.abc import Callable
+from string import Template
+from typing import Dict, List, Optional, Tuple, Union, Any, Type
+
+import requests
+import torch
+from transformers import pipeline, AutoModelForSeq2SeqLM
+
+from haystack import MultiLabel
+from haystack.errors import OpenAIError, OpenAIRateLimitError
+from haystack.modeling.utils import initialize_device_settings
+from haystack.nodes.base import BaseComponent
+from haystack.schema import Document
+
+logger = logging.getLogger(__name__)
+
+
+class BaseTemplate(BaseComponent):
```

Comment:
@vblagoje @ZanSara Should this be called `BasePromptTemplate` instead of `BaseTemplate` maybe?

Reply:
Otherwise users might get confused and think it has anything to do with string.Template, which we import too.